Description: Fix HTMLPurifier loading for versions >= 4.7.0
Author: Markus Frosch <lazyfrosch@debian.org>
Forwarded: https://dev.icinga.org/issues/10652
Last-Update: 2015-11-16
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/modules/monitoring/application/views/helpers/PluginOutput.php
+++ b/modules/monitoring/application/views/helpers/PluginOutput.php
@@ -92,8 +92,15 @@
     {
         if (self::$purifier === null) {
             require_once 'HTMLPurifier/Bootstrap.php';
-            require_once 'HTMLPurifier/HTMLPurifier.php';
-            require_once 'HTMLPurifier/HTMLPurifier.autoload.php';
+            try {
+                require_once 'HTMLPurifier.php';
+                require_once 'HTMLPurifier.autoload.php';
+
+            } catch (ErrorException $e) {
+                // loading failed, try to load old location
+                require_once 'HTMLPurifier/HTMLPurifier.php';
+                require_once 'HTMLPurifier/HTMLPurifier.autoload.php';
+            }
 
             $config = HTMLPurifier_Config::createDefault();
             $config->set('Core.EscapeNonASCIICharacters', true);
